src/state.rs

```diff
@@ -50,4 +59,33 @@ pub const STATE: Item<State> = Item::new("state");
 
 pub const CONFIG: Item<Config> = Item::new("config");
 
+
+// off chain details
+
+pub const OFF_CHAIN_STATE: Item<OffChainState> = Item::new(constants::OFF_CHAIN_STATE);
+
+pub const OFF_CHAIN_VALIDATOR_IDX_MAPPING: Map<&Addr, u16> =
+    Map::new(constants::OFF_CHAIN_VALIDATOR_IDX_MAPPING);
+
+pub const OFF_CHAIN_TIMESTAMP_DETAILS: Map<U64Key, OffchainTimestampDetails> =
+    Map::new(constants::OFFCHAIN_TIMESTAMP_DETAILS);
+
+pub const OFF_CHAIN_STATE_FOR_VALIDATOR: Map<(U64Key, Addr), OffChainValidatorMetrics> =
+    Map::new(constants::OFF_CHAIN_METRICS_FOR_VALIDATOR);
+
+pub const OFF_CHAIN_TIMESTAMPS: Item<OffChainTimestamps> = Item::new("off_chain_timestamps");
```

Comment:
If I were you, I would change this to be a map as well. The key can be auto-incrementing (or maintained in the state). 
<img width="1081" alt="Screen Shot 2021-12-28 at 7 34 24 PM" src="https://user-images.githubusercontent.com/19915413/147574186-fd05dda6-28a3-4dc5-a76a-dd04cda0835c.png">

Reply:
MAP IT ALLLLLL!